### DISCUSSION TITLE
Regression issue?

### DISCUSSION THREAD

**Discussion by skaunov** [noir-lang/noir#2002] [Q&A] [answered]
I'm very new and going through <https://noir-by-example.org/examples/comptime/>. This particular one gave unexpected result, and I'm a bit lost where it's better to report among repos.

```noir
fn main() -> pub Field {
    let known_at_compile_time: comptime Field = 1;

    // let inferred_known = 2;

    let mut bad_var: comptime Field = 3;
    bad_var = 4;

    // known_at_compile_time + inferred_known
    known_at_compile_time + bad_var
}

#[test]
fn test_main() {
    assert(main() == 5);
}
```

```console
devbox@pop-os:~/projects/hello_world$ nargo test --show-output
Running 1 test functions...
Testing test_main...
ok
All tests passed
```

Ran both with plain `noirup` and today nightly. AFAIU https://noir-lang.org/language_concepts/mutability#comptime-values it should return error instead.

**Comment by Savio-Sou** [comment]
Seems that it's behaving as it should be (i.e. 1 + 4 = 5)?

**Reply by skaunov** [reply]
If only subject of that page was arithmetics. =(

I was trying to say that 
```noir
let mut bad_var: comptime Field = 3;
    bad_var = 4;
```
worked, while the page author claims that it was failing at the time of writing.

**Reply by Savio-Sou** [reply]
Ah in that case probably best to feedback to the project's author / original repo 🙌

**Reply by TomAFrench** [reply]
@Savio-Sou this contradicts our [own docs]( https://noir-lang.org/language_concepts/mutability#comptime-values) as well so we'll want to update these.

**Comment by TomAFrench** [comment, marked as answer]
I think this may be an error in the docs as `bad_var`'s value is known for every step of program execution at compile time (the definition of a `comptime` variable).

A better example would be if you had a program similar to below

```rust
fn main(runtime_var: Field) -> pub Field {
    let known_at_compile_time: comptime Field = 1;

    // let inferred_known = 2;

    let bad_var: comptime Field = runtime_var;
    
    // known_at_compile_time + inferred_known
    known_at_compile_time + bad_var
}
```

As `runtime_var` is a argument to the circuit it cannot be known at compile time and so assigning it to a `comptime` variable should fail. A circuit's arguments is the only way in which non-comptime variables can enter the circuit (excluding brillig foreign calls) so in your example of a test (which has no arguments) then all of the circuit's variables will be `comptime`.
